libsql-server/src/http/user/db_factory.rs

```diff
@@ -64,10 +80,37 @@ pub fn namespace_from_headers(
     }
 }
 
+pub fn namespace_from_path(
+    parts: &Parts,
+    disable_namespaces: bool,
+) -> crate::Result<NamespaceName> {
+    if disable_namespaces {
+        return Ok(NamespaceName::default());
+    }
+    if let Some(ns) = extract_namespace(parts) {
+        trace!("extracted namespace from path: {}", ns);
+        return Ok(NamespaceName::from_string(ns)?);
+    }
+    Err(Error::InvalidPath(
+        "/dev path should be in the format /dev/<namespace> and contain only alphanumerics".into(),
+    ))
+}
+
 fn split_namespace(host: &str) -> crate::Result<NamespaceName> {
     let (ns, _) = host.split_once('.').ok_or_else(|| {
         Error::InvalidHost("host header should be in the format <namespace>.<...>".into())
     })?;
     let ns = NamespaceName::from_string(ns.to_owned())?;
     Ok(ns)
 }
+
+fn extract_namespace(parts: &Parts) -> Option<String> {
```

Comment:
there is probably an axum way to extract this namespace from the parts, but I could not figure out how to do it

Reply:
it's simpler than it seems: you need implement `Deserialize` for `NamespaceName`:
```rust
impl<'de> Deserialize for NamespaceName<'de> {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: serde::Deserializer<'de> {
            struct V;

            impl<'de> Visitor<'de> for V {
                type Value = NamespaceName;

                fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
                    write!(f, "a valid namespace name");
                }

                fn visit_string<E>(self, v: String) -> Result<Self::Value, E>
                where
                    E: serde::de::Error, {
                        NamespaceName::from_string(v).map_err(|e| E::custom(e))
                }
            }
        }
}
```

when you have that, you can use the `Path` extractor: https://docs.rs/axum/latest/axum/extract/struct.Path.html

Reply:
Thank you so much, I shall try this 🙏